atc/steps.go

```diff
@@ -90,6 +91,10 @@ func (step *Step) UnmarshalJSON(data []byte) error {
 		return ErrNoStepConfigured
 	}
 
+	if !coreStepDeclared {
+		return errors.New("no core step type declared (e.g. get, put, task, etc.)")
```

Comment:
Can we pull this out as an `ErrNoCoreStepDeclared` var? It's just nice to see the error conditions defined somewhere more discoverable rather than inlined.